simple-facebook-instant-articles.php

```diff
@@ -494,17 +532,37 @@ public function ad_targeting_js() {
 	}
 
 	/**
-	 * Generates HTML string for DOM node object.
-	 *
-	 * @param DOMNode $node Node object to generate the HTML string for.
-	 *
-	 * @return string       HTML string/markup for supplied DOM node.
+	 * Generates XML/HTML string for DOM node object.
+	 *
+	 * @param DOMNode $node         Node object to generate the XML/HTML string for.
+	 * @param string  $save_as      Whether to save Node as XML or not.
+	 *                              Default - true, save node as XML.
+	 *                              This is important(!):
+	 *                              1) Final post content should be returned as HTML -
+	 *                              otherwise <script></script> tags will be output as <script />
+	 *                              which is incorrect HTML.
+	 *                              2) However, while working with DOM document fragments we need XML,
+	 *                              otherwise nodes are not generated correctly; and never replaced.
+	 * @param bool    $is_self_html Whether to include node's markup itself in output.
+	 *                              Default - false, only return children markup.
+	 *
+	 * @return string               XML/HTML string/markup for supplied DOM node.
 	 */
-	protected function get_html_for_node( \DOMNode $node ) {
+	protected function get_html_for_node( \DOMNode $node, $save_as = 'xml', $is_self_html = false ) {
```

Comment:
3. If we need to handle XML or HTML - I suggest creating separate methods. Keep things as simple as possible. I think the code by itself should be as readable as possible without having to havelots of inline docs.